### ISSUE TITLE
[FIO blockchain protobuf] Change refBlockPrefix type from int to uint64

### ISSUE THREAD

**Issue by Defuera** [trustwallet/wallet-core#878]
**Describe the bug**
It's needed to change refBlockPrefix type from int to uint64, now we are getting overflow in java code

**Comment by hewigovens** [comment]
can you share your refBlockPrefix? I have a quick pr but it's still encoded as uint32_t

**Comment by catenocrypt** [comment]
I could change the type to larger in walletcore code, but it is still encoded in 32 bits in the ABI, we cannot change that.
Maybe try taking mod 2**32 before supplying the value (&& 0xFFFFFFFF).
The prefix values I have seen fit in 32 bits, e.g. dec 2523882092 hex 0x966F626C.

```shell
curl -d '{"block_num_or_id": "123020"}' http://testnet.fioprotocol.io/v1/chain/get_block
```

See reference code in Fio JS SDK:
https://github.com/fioprotocol/fiojs/blob/master/src/transaction.abi.json

```
                {
                    "name": "ref_block_num",
                    "type": "uint16"
                },
                {
                    "name": "ref_block_prefix",
                    "type": "uint32"
                },
```

**Comment by hewigovens** [comment]
so this is actually a protobuf java issue, the type uint32_t generated should be `long` not `int`

**Comment by hewigovens** [comment]
in this case, `static_cast<uint32_t>(refBlockPrefix)` should be safe @catenocrypt

**Comment by catenocrypt** [comment]
Merged with PR #879 .
